polaris/evaluate/utils.py

```diff
@@ -3,85 +3,71 @@
-        or input_values[test_label].get(target_label) is None
+        or input_values.predictions.get(test_label) is None
+        or input_values.predictions[test_label].get(target_label) is None
     ):
         return None
     else:
-        return input_values[test_label][target_label][mask]
+        return input_values.predictions[test_label][target_label][mask]
 
 
 def evaluate_benchmark(
     target_cols: list[str],
     metrics: list[Metric],
-    y_true: PredictionsType,
-    y_pred: Optional[PredictionsType] = None,
-    y_prob: Optional[PredictionsType] = None,
+    y_true: IncomingPredictionsType,
+    y_pred: Optional[IncomingPredictionsType] = None,
+    y_prob: Optional[IncomingPredictionsType] = None,
 ):
-    y_true = normalize_predictions_type(y_true, target_cols)
-    y_pred = normalize_predictions_type(y_pred, target_cols)
-    y_prob = normalize_predictions_type(y_prob, target_cols)
+    y_true = BenchmarkPredictions(predictions=y_true, target_cols=target_cols)
+    if y_pred is not None:
+        y_pred = BenchmarkPredictions(predictions=y_pred, target_cols=target_cols)
+    if y_prob is not None:
+        y_prob = BenchmarkPredictions(predictions=y_prob, target_cols=target_cols)
 
-    if y_pred and set(y_true.keys()) != set(y_pred.keys()):
-        raise KeyError(f"Missing keys for at least one of the test sets. Expecting: {sorted(y_true.keys())}")
+    if y_pred and set(y_true.predictions.keys()) != set(y_pred.predictions.keys()):
+        raise KeyError(
+            f"Missing keys for at least one of the test sets. Expecting: {sorted(y_true.predictions.keys())}"
+        )
 
     # Results are saved in a tabular format. For more info, see the BenchmarkResults docs.
     scores: ResultsType = pd.DataFrame(columns=BenchmarkResults.RESULTS_COLUMNS)
 
     # For every test set...
-    for test_label, y_true_subset in y_true.items():
+    for test_label, y_true_test in y_true.predictions.items():
         # For every metric...
         for metric in metrics:
             if metric.is_multitask:
                 # Multi-task but with a metric across targets
                 score = metric(
-                    y_true=y_true_subset, y_pred=y_pred.get(test_label), y_prob=y_prob.get(test_label)
+                    y_true=y_true_test,
+                    y_pred=y_pred.predictions.get(test_label),
+                    y_prob=y_prob.predictions.get(test_label),
                 )
 
                 scores.loc[len(scores)] = (test_label, "aggregated", metric, score)
                 continue
 
-            if not isinstance(y_true_subset, dict):
+            if not isinstance(y_true_test, dict):
                 # Single task
                 score = metric(
-                    y_true=y_true_subset, y_pred=y_pred.get(test_label), y_prob=y_prob.get(test_label)
+                    y_true=y_true_test,
+                    y_pred=y_pred.predictions.get(test_label),
+                    y_prob=y_prob.predictions.get(test_label),
```

Comment:
**Question**: y_pred and y_prob may be None, right? In which case this would crash?